hopsworks-IT/src/test/java/io/hops/hopsworks/WebDriverFactory.java

```diff
@@ -0,0 +1,177 @@
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang3.SystemUtils;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.firefox.FirefoxOptions;
+import org.openqa.selenium.firefox.GeckoDriverService;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.rauschig.jarchivelib.Archiver;
+import org.rauschig.jarchivelib.ArchiverFactory;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.io.FilenameUtils;
+import org.openqa.selenium.chrome.ChromeDriverService;
+
+public class WebDriverFactory {
+
+  private static final Logger LOGGER = Logger.getLogger(WebDriverFactory.class.getName());
+  private static final String GECKODRIVER_VERSION = "0.23.0";
+  private static final String CHROMEDRIVER_VERSION = "2.42";
+  private static final String GECKODRIVER = "geckodriver";
+  private static final String CHROMEDRIVER = "chromedriver";
+  private static final String GECKODRIVER_URL = "https://github.com/mozilla/geckodriver/releases/download/v"
+      + GECKODRIVER_VERSION + "/geckodriver-v" + GECKODRIVER_VERSION + "-";
+  private static final String CHROMEDRIVER_URL = "https://chromedriver.storage.googleapis.com/" + CHROMEDRIVER_VERSION
+      + "/chromedriver_";
+  private static final int SUPPORTED_CHROME_VERSION = 68;
+  private static final int SUPPORTED_FIREFOX_VERSION = 57;
+  private static final String BROWSER_ENV = "BROWSER";
+  private static final String BROWSER_UI_ENV = "HEADLESS";
+  private static final String HOPSWORKS_URL_ENV = "HOPSWORKS_URL";
+
+  public static WebDriver getWebDriver() {
+    WebDriver driver = null;
+    String downLoadsDir = FileUtils.getTempDirectory().toString() + "/webDrivers/";
+    downloadDrivers(downLoadsDir);
+    String chosenBrowser = System.getenv(BROWSER_ENV);
+
+    if ("firefox".equalsIgnoreCase(chosenBrowser)) {
+      File geekoDriver = new File(downLoadsDir + GECKODRIVER);
+      if (geekoDriver.exists()) {
+        System.setProperty(GeckoDriverService.GECKO_DRIVER_EXE_PROPERTY, downLoadsDir + GECKODRIVER);
+        FirefoxOptions firefoxOptions = new FirefoxOptions();
+        if ("true".equals(System.getenv(BROWSER_UI_ENV))) {
+          firefoxOptions.addArguments("--headless");
+        }
+        driver = new FirefoxDriver(firefoxOptions);
+      }
+    }
+
+    if (driver == null) {
+      File chromeDriver = new File(downLoadsDir + CHROMEDRIVER);
+      if (chromeDriver.exists()) {
+        System.setProperty(ChromeDriverService.CHROME_DRIVER_EXE_PROPERTY, downLoadsDir + CHROMEDRIVER);
+        ChromeOptions chromeOptions = new ChromeOptions();
+        if ("true".equals(System.getenv(BROWSER_UI_ENV))) {
+          chromeOptions.addArguments("--headless");
```

Comment:
```suggestion
          chromeOptions.addArguments("--headless");
          chromeOptions.addArguments("--no-sandbox");
          chromeOptions.addArguments("--disable-dev-shm-usage");
```

Reply:
`--no-sandbox` and `--disable-dev-shm-usage` are required to run the tests on jenkins